provisioning/baremetal_pod_test.go

```diff
@@ -99,16 +98,16 @@ func TestNewMetal3InitContainers(t *testing.T) {
 			config: managedProvisioning().build(),
 			expectedContainers: []corev1.Container{
 				{
-					Name:  "metal3-ipa-downloader",
-					Image: images.IpaDownloader,
+					Name:  "metal3-static-ip-set",
+					Image: images.StaticIpManager,
 				},
 				{
-					Name:  "metal3-machine-os-downloader",
-					Image: images.MachineOsDownloader,
+					Name:  "machine-os-images",
+					Image: images.IronicAgent,
```

Comment:
This doesn't look right - machine-os-images does not use the IronicAgent image, it uses the MachineOSImages image.

It appears the test is only checking the number of containers... I think we should test the contents or just put the number here.

Reply:
I actually test the names of the containers, although I didn't add the test here, so the names are correct
I probably overlooked the Image name by mistake, it was not intentional
I'm going to fix this and maybe we can add the check on the containers names in a different PR